hatchery/nextflow.go

```diff
@@ -1075,11 +1082,33 @@ func setupSubnet(subnetName string, cidr string, vpcid string, ec2Svc *ec2.EC2)
 		return exsubnet.Subnets[0].SubnetId, nil
 	}
 
+	// Fetch all availability zones
+	// this is being limited to a region by the ec2svc that gets passed in.
+	describeZonesOutput, err := ec2Svc.DescribeAvailabilityZones(&ec2.DescribeAvailabilityZonesInput{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to describe availability zones: %v", err)
+	}
+
+	// Avoid us-east-1e as it is technically neglected, and does not have modern instance types.
+	// Find a suitable availability zone that is not us-east-1e
+	var selectedZone string
+	for _, zone := range describeZonesOutput.AvailabilityZones {
+		if *zone.State == "available" && !strings.EqualFold(*zone.ZoneName, "us-east-1e") {
```

Comment:
One super small suggestion I could give is, since *zone.State is always `available` (as per the [docs](https://docs.aws.amazon.com/sdk-for-go/api/service/ec2/#AvailabilityZone)), we can change the order of the `if` condition to use short circuit eval a little better

```
if !strings.EqualFold(*zone.ZoneName, "us-east-1e") && *zone.State == "available"
```